src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryFormHelper.ts

```diff
@@ -0,0 +1,202 @@
+    }
+  }
+
+  return {
+    varFile: {
+      identifier: prevStepData?.varFile?.identifier || '',
+      type: TerraformStoreTypes.Remote,
+      spec: {
+        store: {
+          spec: {
+            repositoryName: prevStepData?.varFile?.spec?.store?.spec?.repositoryName || '',
+            artifacts: prevStepData?.varFile?.spec?.store?.spec?.artifacts || defaultArtifacts
+          }
+        }
+      }
+    }
+  }
+}
+
+export const getConnectorRef = (isConfig: boolean, isTerraformPlan: boolean, prevStepData: any) => {
+  let connectorValue
+  if (isConfig && isTerraformPlan) {
+    connectorValue = prevStepData?.formValues.spec?.configuration?.configFiles?.store?.spec?.connectorRef
+  } else if (isConfig) {
+    connectorValue = prevStepData?.formValues?.spec?.configuration?.spec?.configFiles?.store?.spec?.connectorRef
+  } else connectorValue = prevStepData?.varFile?.spec?.store?.spec?.connectorRef
+
+  if (size(prevStepData?.identifier) > 5) {
+    return prevStepData?.identifier
+  }
+
+  if (!isUndefined(connectorValue?.connector?.identifier)) {
+    return connectorValue.connector.identifier
+  }
+
+  return connectorValue
+}
+
+export const terraformArtifactorySchema = (isConfig: boolean, getString: any) => {
+  const artifacts = {
+    repositoryName: Yup.string().required(getString('cd.artifactFormErrors.repositoryName')),
+    artifacts: Yup.array()
+      .of(
+        Yup.object().shape({
+          artifactFile: Yup.object().shape({
+            artifactPathExpression: Yup.string().required(getString('cd.pathCannotBeEmpty')),
+            name: Yup.string().required(getString('cd.artifactFormErrors.artifactName'))
+          })
+        })
+      )
+      .required()
+  }
+
+  const configSetup = {
+    configFiles: Yup.object().shape({
+      store: Yup.object().shape({
+        spec: Yup.object().shape({
+          ...artifacts
+        })
+      })
+    })
+  }
+
+  if (isConfig) {
+    return Yup.object().shape({
+      spec: Yup.object().shape({
+        configuration: Yup.object().shape({
+          ...configSetup
+        })
+      })
+    })
+  }
+
+  return Yup.object().shape({
+    varFile: Yup.object().shape({
+      identifier: Yup.string().required(getString('common.validation.identifierIsRequired')),
+      spec: Yup.object().shape({
+        store: Yup.object().shape({
+          spec: Yup.object().shape({
+            ...artifacts
+          })
+        })
+      })
+    })
+  })
+}
+
+export const tfArtifactoryFormInputNames = (isConfig: boolean) => {
+  if (isConfig) {
+    return {
+      repositoryName: 'spec.configuration.configFiles.store.spec.repositoryName',
+      artifacts: 'spec.configuration.configFiles.store.spec.artifacts'
+    }
+  }
+
+  return {
+    repositoryName: 'varFile.spec.store.spec.repositoryName',
+    artifacts: 'varFile.spec.store.spec.artifacts'
+  }
+}
+/* istanbul ignore next */
+export const formatOnSubmitData = (values: any, prevStepData: any, connectorValue: any) => {
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232248)
```suggestion
export const formatOnSubmitData = (values: unknown, prevStepData: any, connectorValue: any) => {
```